### PR TITLE
Chore: issue a warning if __version__ can't be imported

### PR DESCRIPTION
**Pull request by GeorgeSittas**

### CODE REVIEW

File: sqlglot/__init__.py [hunk 1]
```diff
@@ -49,11 +50,13 @@
...
 
+logger = logging.getLogger("sqlglot")
+
 
 try:
-    from sqlglot._version import __version__, __version_tuple__  # type: ignore
+    from sqlglot._version import __version__, __version_tuple__
 except ImportError:
-    pass
+    logger.warning("Unable to set __version__, consider installing SQLGlot first.")
```
[the remark this answers is not in the document]
Comment (tobymao) [reply sqlglot/__init__.py:59]: and don't make this a warning, make it an error

[the remark this answers is not in the document]
Comment (GeorgeSittas) [reply sqlglot/__init__.py]: done

[the remark this answers is not in the document]
Comment (GeorgeSittas) [reply sqlglot/__init__.py]: @tobymao I wonder whether this message will always be printed when using sqlglot as a non-developer. For example, if you `import sqlglot` or use its CLI as `python -m sqlglot ...`, do we want it to log that?

[the remark this answers is not in the document]
Comment (GeorgeSittas) [reply sqlglot/__init__.py]: Seems like it is printed regardless:
```python
➜  sqlglot git:(jo/warn_about_version) python -m sqlglot "select foo"
Unable to set __version__, run `pip install -e .` or `python setup.py develop` first.
SELECT
  "foo"
```

This is without having a `_version.py` file at all.

**Pull request merged**